packages/rum/src/viewCollection.ts

```diff
@@ -26,26 +33,33 @@ export interface ViewMeasures {
   userActionCount: number
 }
 
+export enum ViewLoadType {
+  INITIAL_PAGE_LOAD = 'initial page load',
+  ROUTE_CHANGE = 'route change',
+}
+
 export const THROTTLE_VIEW_UPDATE_PERIOD = 3000
 
 export function startViewCollection(location: Location, lifeCycle: LifeCycle, session: RumSession) {
   let currentLocation = { ...location }
   const startOrigin = 0
-  let currentView = newView(lifeCycle, currentLocation, session, startOrigin)
+  let currentViewLoadType: ViewLoadType = ViewLoadType.INITIAL_PAGE_LOAD
```

Comment:
why do we need to keep a reference to the currentViewLoadType?
Could we just call newView with the according type depending on the case?

Reply:
I guess I could use `currentView.loadType` to renew the view with the proper loading type instead of keeping this reference